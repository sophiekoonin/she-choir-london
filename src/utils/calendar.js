const { GCAL_API_KEY, GCAL_ID } = process.env
const calendarUrl = `https://www.googleapis.com/calendar/v3/calendars/${GCAL_ID}/events?key=${GCAL_API_KEY}`

export const getEvents = async () => {
  const response = await fetch(calendarUrl)
  const { items } = await response.json()
  return items
    ? items.map(event => ({
        start: event.start.date || event.start.dateTime,
        end: event.end.date || event.end.dateTime,
        title: event.summary,
        location: event.location
      }))
    : []
}
