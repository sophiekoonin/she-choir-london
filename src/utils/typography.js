import Typography from 'typography'
import { MOBILE_MEDIA_QUERY } from 'typography-breakpoint-constants'

const theme = {
  title: 'SHE Choir',
  baseFontSize: '18px',
  baseLineHeight: 1.5625,
  googleFonts: [
    {
      name: 'Raleway Dots',
      styles: ['400']
    },
    {
      name: 'Raleway',
      styles: ['400, 600']
    },
    {
      name: 'Source Serif Pro',
      styles: ['600']
    },
    {
      name: 'Source Sans Pro',
      styles: ['400', '400i', '700']
    }
  ],
  headerFontFamily: ['Source Serif Pro', 'sans-serif'],
  bodyFontFamily: ['Source Sans Pro', 'sans-serif'],
  headerColor: 'hsla(0,0%,0%,1)',
  bodyColor: 'hsla(0,0%,0%,0.8)',
  headerWeight: 600,
  bodyWeight: 400,
  boldWeight: 700,
  overrideStyles: ({ adjustFontSizeTo, scale, rhythm }, options) => ({
    a: {
      color: '#fb251b',
      textDecoration: 'underline'
    },
    'a:hover,a:active': {
      color: options.bodyColor
    },
    'h1,h2,h3,h4,h5,h6': {
      marginTop: rhythm(2)
    },
    blockquote: {
      ...scale(1 / 5),
      color: 'rgba(0,0,0,0.4)',
      paddingLeft: rhythm(18 / 16),
      marginLeft: 0,
      borderLeft: `${rhythm(6 / 16)} solid`,
      borderColor: '#fcea0e'
    },
    'blockquote > :last-child': {
      marginBottom: 0
    },
    'blockquote cite': {
      ...adjustFontSizeTo(options.baseFontSize),
      color: options.bodyColor,
      fontWeight: options.bodyWeight
    },
    'blockquote cite:before': {
      content: '"â€” "'
    },
    [MOBILE_MEDIA_QUERY]: {
      blockquote: {
        marginLeft: rhythm(-3 / 4),
        marginRight: 0,
        borderLeft: `${rhythm(3 / 16)} solid`,
        borderColor: '#fcea0e',
        paddingLeft: rhythm(9 / 16)
      }
    }
  })
}
const typography = new Typography(theme)

export default typography
