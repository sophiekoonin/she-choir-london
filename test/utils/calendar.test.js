import { getEvents } from '../../src/utils/calendar'

const response = {
  kind: 'calendar#events',
  etag: '"adsfasdf"',
  summary: 'SHE Choir London Gigs',
  updated: '2018-07-16T19:02:34.633Z',
  timeZone: 'Europe/London',
  accessRole: 'reader',
  defaultReminders: [],
  nextSyncToken: 'asdfdasf==',
  items: [
    {
      kind: 'calendar#event',
      etag: '"asdfadsf"',
      id: 'mylovelyid',
      status: 'confirmed',
      htmlLink: 'https://www.example.com',
      created: '2016-08-30T15:49:01.000Z',
      updated: '2016-12-12T16:08:44.359Z',
      summary: 'Super exciting event name',
      location: 'West Covina, California',
      creator: {
        email: 'someone@example.com',
        displayName: 'SHE Choir London'
      },
      organizer: {
        email: 'calendar@id',
        displayName: 'SHE Choir London Gigs',
        self: true
      },
      start: {
        dateTime: '2016-01-12T18:30:00Z'
      },
      end: {
        dateTime: '2016-01-12T19:30:00Z'
      }
    },
    {
      kind: 'calendar#event',
      etag: '"asdfadsf"',
      id: 'mylovelyid1',
      status: 'confirmed',
      htmlLink: 'https://www.example.com',
      created: '2016-08-30T15:49:01.000Z',
      updated: '2016-12-12T16:08:44.359Z',
      summary: 'Another event',
      location: 'Mordor',
      creator: {
        email: 'someone@example.com',
        displayName: 'SHE Choir London'
      },
      organizer: {
        email: 'calendar@id',
        displayName: 'SHE Choir London Gigs',
        self: true
      },
      start: {
        dateTime: '2016-03-13T18:30:00Z'
      },
      end: {
        dateTime: '2016-03-13T19:30:00Z'
      }
    },
    {
      kind: 'calendar#event',
      etag: '"asdfadsf"',
      id: 'mylovelyid2',
      status: 'confirmed',
      htmlLink: 'https://www.example.com',
      created: '2016-08-30T15:49:01.000Z',
      updated: '2016-12-12T16:08:44.359Z',
      summary: 'Something else',
      location: 'Alderaan',
      creator: {
        email: 'someone@example.com',
        displayName: 'SHE Choir London'
      },
      organizer: {
        email: 'calendar@id',
        displayName: 'SHE Choir London Gigs',
        self: true
      },
      start: {
        dateTime: '2016-05-14T18:30:00Z'
      },
      end: {
        dateTime: '2016-05-14T19:30:00Z'
      }
    }
  ]
}

describe('calendar', () => {
  it('should map the response items into a slimmed-down array of events', async () => {
    fetch.once(JSON.stringify(response))

    const events = await getEvents()
    expect(events).toMatchSnapshot()
  })

  it('should return an empty array if the fetch fails', async () => {
    fetch.once(JSON.stringify({ error: 'oh no' }))
    const events = await getEvents()

    expect(events).toEqual([])
  })
})
