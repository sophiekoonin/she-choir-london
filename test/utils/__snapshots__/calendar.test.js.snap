// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar should map the response items into a slimmed-down array of events 1`] = `
Array [
  Object {
    "end": "2016-01-12T19:30:00Z",
    "location": "West Covina, California",
    "start": "2016-01-12T18:30:00Z",
    "title": "Super exciting event name",
  },
  Object {
    "end": "2016-03-13T19:30:00Z",
    "location": "Mordor",
    "start": "2016-03-13T18:30:00Z",
    "title": "Another event",
  },
  Object {
    "end": "2016-05-14T19:30:00Z",
    "location": "Alderaan",
    "start": "2016-05-14T18:30:00Z",
    "title": "Something else",
  },
]
`;
